content-src/components/DiscoveryStreamComponents/DSEmptyState/DSEmptyState.jsx

```diff
@@ -1,12 +1,84 @@
+import {actionCreators as ac, actionTypes as at} from "common/Actions.jsm";
 import React from "react";
 
 export class DSEmptyState extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.onReset = this.onReset.bind(this);
+    this.state = {};
+  }
+
+  componentWillUnmount() {
+    if (this.timeout) {
+      clearTimeout(this.timeout);
+    }
+  }
+
+  onReset() {
+    if (this.props.dispatch && this.props.feed) {
+      const {feed} = this.props;
+      const {url} = feed;
+      this.props.dispatch({
+        type: at.DISCOVERY_STREAM_FEED_UPDATE,
+        data: {
+          feed: {
+            ...feed,
+            data: {
+              ...feed.data,
+              status: "waiting",
+            },
+          },
+          url,
+        },
+      });
+
+      this.setState({waiting: true});
+      this.timeout = setTimeout(() => {
+        this.timeout = null;
+        this.setState({
+          waiting: false,
+        });
+      }, 300);
+
+      this.props.dispatch(ac.OnlyToMain({type: at.DISCOVERY_STREAM_RETRY_FEED, data: {feed}}));
+    }
+  }
+
+  renderButton() {
+    if (this.props.status === "waiting" || this.state.waiting) {
+      return (
+        <button className="try-again-button waiting">Loading...</button>
```

Comment:
Do these strings need to be variables for localization?

Reply:
There are more strings in this file that are hard coded, I think if this is an issue, we can punt it to another bug.

Reply:
The other english string was landed for 68 with a comment of "english only for 68 but localized for 69+"
https://github.com/mozilla/activity-stream/pull/4877#discussion_r273171475

So we definitely want a P1 bug if we expect this to be more than just english for 69 (although I'm not sure if that timeline matches up since the original comment).

Reply:
Filed https://bugzilla.mozilla.org/show_bug.cgi?id=1556862